### ISSUE TITLE
whereIn whith more than two thousand records doesn't work

### ISSUE THREAD

**Issue by bioleyl** [yajra/laravel-oci8#590]
I had to make a query with 6000 records in a whereIn clause. Oracle send me an error: ORA-00936: missing expression  

After looking the produced request, I found that on every chunk of 1000 records in the whereIn, the library added an 'or'  

```
// First 1000
id in (1, 2, ..., 1000)
// Second 1000
or id in (1001, 1002, ..., 2000)
// Third 1000
or or id in (2001, 2002, ..., 3000)  
```

The multiple 'or' is added in the file OracleGrammar.php line 549
```
private function resolveClause($column, $values, $type)
    {
        $chunks      = array_chunk($values, 1000);
        $whereClause = '';
        $i           = 0;
        $type        = $this->wrap($column) . ' '.$type.' ';
        foreach ($chunks as $ch) {
            if ($i > 0) { // <---- Here every chunk after the first one will add 'or'
                $type = ' or ' . $type . ' ';
            }
            $whereClause .= $type . '('.implode(', ', $ch).')';
            $i++;
        }

        return '(' . $whereClause . ')';
    }
```
Here is a debug version of the function
```
private function resolveClause($column, $values, $type)
    {
        $chunks      = array_chunk($values, 1000);
        $whereClause = '';
        $i           = 0;
        $type        = $this->wrap($column) . ' '.$type.' ';
        foreach ($chunks as $ch) {
            if ($i === 1) { // <---- Now only the second loop will and an 'or' and no other
                $type = ' or ' . $type . ' ';
            }
            $whereClause .= $type . '('.implode(', ', $ch).')';
            $i++;
        }

        return '(' . $whereClause . ')';
    }
```

I made a PR with the fix: #591

**Comment by yajra** [comment]
I think it's the correct query to add `or` on every chunk because using `and` will yield to wrong results? 

Also the expected query should be wrapped with `()`, making a single `and` where clause.

```
(
id in (1, 2, ..., 1000)
// Second 1000
or id in (1001, 1002, ..., 2000)
// Third 1000
or or id in (2001, 2002, ..., 3000)  
)
```

**Comment by bioleyl** [comment]
Using 'OR' is right and everything inside () is right too. The problem is not here

The bug is on every loop after the first one, 'OR' is concatenated. The first time 'OR' is concatenated is right but after the second loop, it should not continue.

```
// See OracleGrammar.php line 549
// This is not working because multiple 'OR' chained
select * from MYTABLE where (ID in (1000) or ID in (1001) or or ID in (1002) or or or ID in (1003));
//                                      1x right         2x wrong            3x wrong  

// This is the right query. Addind 'OR' to $type only at the second loop
select * from MYTABLE where (ID in (1000) or ID in (1001) or ID in (1002) or ID in (1003));
//                                      1x right       1x right        1x right   
```

**Comment by yajra** [comment]
Oic, thanks for clarifying. Can you confirm that the PR you submitted is working? I will merge and release it when confirmed. Cannot check further right now. Thanks!

**Comment by bioleyl** [comment]
I confirm that this fix has solve the bug. I tested with less than 1000 (1 chunk) between 1000 and 2000 (2 chunk) and more than 2000 (more than 2 chunk).

I didn't write unit tests but everything is working also when you use relations.

**Comment by yajra** [comment]
PR #591  merged and released on v8.0.1. Thanks! 🚀
